CMakeLists.txt

```diff
@@ -18,6 +18,13 @@ project(pybind11)
 set(PYBIND11_MASTER_PROJECT OFF)
 if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
   set(PYBIND11_MASTER_PROJECT ON)
+  set(PYBIND11_EXPORT_NAME "${PROJECT_NAME}Targets")
+  set(PYBIND11_INSTALL_EXPORT ON)
+else ()
+  if (NOT PYBIND11_EXPORT_NAME)
+    set(PYBIND11_EXPORT_NAME "${PROJECT_NAME}Targets")
+  endif ()
```

Comment:
Since `PYBIND11_EXPORT_NAME` is only ever needed in the `if(PYBIND11_INSTALL)` block, I'd move it in there to improve locality. No need for separate `PYBIND11_MASTER_PROJECT` behavior.